Clean up `_add_swift_host_library_single`

compnerd opened the pull request:
<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves SR-NNNN.

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

compnerd committed:
build: remove `MACCATALYST_BUILD_FLAVOR` from `_add_swift_host_library_single`

compnerd committed:
build: remove `DEPLOYMENT_VERSION_MACCATALYST` from `_add_swift_host_library_single`

compnerd committed:
build: remove `IS_SDK_OVERLAY` from `_add_swift_host_library_single`

compnerd committed:
build: remove `IS_STDLIB_CORE` from `_add_swift_host_library_single`

compnerd committed:
build: remove `IS_STDLIB` from `_add_swift_host_library_single`

compnerd committed:
build: remove `TARGET_LIBRARY` from `_add_swift_host_library_single`

compnerd committed:
build: remove `NOSWIFTRT` from `_add_swift_host_library_single`

compnerd committed:
build: remove `DONT_EMBED_BITCODE` from `_add_swift_host_library_single`

compnerd committed:
build: remove `OBJECT_LIBRARY` from `_add_swift_host_library_single`

compnerd committed:
build: remove `MODULE` path from `_add_swift_host_library_single`

compnerd committed:
build: remove `MODULE_TARGETS` from `_add_swift_host_library_single`

compnerd committed:
build: remove `C_COMPILE_FLAGS` from `_add_swift_host_library_single`

compnerd committed:
build: remove `LINK_LIBRARIES` from `_add_swift_host_library_single`

compnerd committed:
build: remove `PRIVATE_LINK_LIBRARIES` from `_add_swift_host_library_single`

compnerd committed:
build: remove `DEPENDS` from `_add_swift_host_library_single`

compnerd committed:
build: remove `FRAMEWORK_DEPENDS` from `_add_swift_host_library_single`

compnerd committed:
build: remove `FRAMEWORK_WEAK_DEPENDS` from `_add_swift_host_library_single`

compnerd committed:
build: remove `LINK_FLAGS` from `_add_swift_host_library_single`

compnerd committed:
build: remove `INCORPORATE_OBJECT_LIBRARIES` from `_add_swift_host_library_single`

compnerd committed:
build: remove `INCORPORATE_OBJECT_LIBRARIES_SHARED_ONLY` from `_add_swift_host_library_single`

compnerd committed:
build: remove `FILE_DEPENDS` from `_add_swift_host_library_single`

compnerd committed:
build: remove `SWIFT_COMPILE_FLAGS` from `_add_swift_host_library_single`

compnerd committed:
build: remove `DEPLOYMENT_VERSION_*` from `_add_swift_host_library_single`

compnerd committed:
build: remove `DARWIN_INSTALL_NAME_DIR` from `_add_swift_host_library_single`

compnerd committed:
build: remove Swift support for `_add_swift_host_library_single`

The host libraries do not build Swift code, these libraries are C++
libraries used for the compiler.

compnerd committed:
build: cleanup unreferenced variable `SWIFTLIB_SINGLE_EXTERNAL_SOURCES` in `_add_swift_host_library_single`

compnerd committed:
build: simplify `INSTALL_NAME_DIR` computation in `_add_swift_host_library_single`

compnerd committed:
build: remove some un-valuved variables in `_add_swift_host_library_single`

compnerd committed:
build: remove unreferenced variable in `_add_swift_host_library_single`

compnerd committed:
build: remove ICU references in `_add_swift_host_library_single`

compnerd committed:
build: remove swiftCore workaround in `_add_swift_host_library`

compnerd committed:
build: inline some variables in `_add_swift_host_library_single`

compnerd commented:
CC: @xiaobai @gottesmm @drexin

compnerd commented:
@karwa hopefully this explains somewhat the reason for the split - this allows us to nearly remove all the options

compnerd commented:
@swift-ci please clean test

compnerd commented:
@swift-ci please test Windows platform

drexin commented on cmake/modules/AddSwift.cmake:
```diff
@@ -674,26 +674,12 @@ endfunction()
-#     [C_COMPILE_FLAGS flag1...]
-#     [SWIFT_COMPILE_FLAGS flag1...]
-#     [LINK_FLAGS flag1...]
-#     [FILE_DEPENDS target1 ...]
```
`add_swift_host_library` still defines `FILE_DEPENDS` as one of the parameters, but doesn't use it anywhere, so I'm wondering if it should be removed there as well.

compnerd replied:
Hmm, perhaps I missed it?  It should be removed in the final view of things.  The individual steps are meant to be easier to digest.

compnerd replied:
Hmm, it hasn't supported that for a while, it did however document it as supporting it before this set of changes.  Fake news! :-p

compnerd commented:
@swift-ci please clean test

compnerd commented:
@swift-ci please test Windows platform

compnerd commented:
@swift-ci please test Windows platform

compnerd commented:
@swift-ci please clean test Windows platform

compnerd commented:
@swift-ci please test macOS platform

compnerd commented:
@swift-ci please clean test macOS platform

compnerd commented:
@swift-ci please test Windows platform

compnerd commented:
@swift-ci please test Windows platform

compnerd commented:
@swift-ci please clean test

compnerd commented:
@swift-ci please clean test

compnerd commented:
@swift-ci please test Windows platform

compnerd commented:
@swift-ci please test Windows platform

compnerd commented:
@swift-ci please clean test

compnerd commented:
@swift-ci please clean test

compnerd commented:
@swift-ci please clean test

compnerd commented:
@swift-ci please clean test macOS platform

compnerd committed:
build: use `_add_swift_target_library_single`

This library is meant to run on the platform where the testing occurs,
requiring bits to run.  This makes this library a target library.

compnerd commented:
@swift-ci please test Linux platform

compnerd commented:
@swift-ci please clean test

compnerd commented:
@swift-ci please test Windows platform

compnerd commented:
@swift-ci please test Windows platform
